src/config.rs

```diff
@@ -22,3 +22,37 @@ pub(crate) fn prefer_dark() -> bool {
 
     matches!(stdout, Some(s) if s.trim().ends_with("uint32 1"))
 }
+
+/// Query system configuration for buttons layout.
+/// Should be updated to use standard xdg-desktop-portal specs once available
+/// https://github.com/flatpak/xdg-desktop-portal/pull/996
+pub(crate) fn get_button_layout_config() -> Option<(String, String)> {
+    let config_string = Command::new("dbus-send")
+        .arg("--reply-timeout=100")
+        .arg("--print-reply=literal")
+        .arg("--dest=org.freedesktop.portal.Desktop")
+        .arg("/org/freedesktop/portal/desktop")
+        .arg("org.freedesktop.portal.Settings.Read")
+        .arg("string:org.gnome.desktop.wm.preferences")
+        .arg("string:button-layout")
+        .output()
+        .ok()
+        .and_then(|out| String::from_utf8(out.stdout).ok());
+
+    let config_string = config_string.unwrap();
```

Comment:
I'm not sure if crashing in this case is right, could we return `None` using `?` instead? just in case.
```suggestion
        .and_then(|out| String::from_utf8(out.stdout).ok())?;
```

Reply:
Good catch, it wasn't intended